### PR TITLE
Clearing cache of projects that are not in the current solution

### PR DESCRIPTION
**Pull request by TIHan**
This will help clear caches from both VS and FCS when a project doesn't exist in the current solution.

### CODE REVIEW

File: src/fsharp/service/service.fs [hunk 1]
```diff
@@ -846,6 +846,12 @@ type BackgroundCompiler(legacyReferenceResolver, projectCacheSize, keepAssemblyC
                 if startBackgroundCompileIfAlreadySeen then 
                    bc.CheckProjectInBackground(options, userOpName + ".StartBackgroundCompile"))
 
+    member bc.ClearCache(options : FSharpProjectOptions list, userOpName) =
```
Comment (cartermp) [inline src/fsharp/service/service.fs:849]: if this is `seq<FSharpProjectOptions>` then you won't need to convert the dictionary into a list on the VS side

Comment (TIHan) [reply src/fsharp/service/service.fs:849]: I think it makes sense, considering FSharpProjectOptions is considered immutable.

Comment (KevinRansom) [reply src/fsharp/service/service.fs:849]: since it just iterates through the collection, seq should be sufficient typing.

**Comment by TIHan** [comment]
Yes, that is right. :) It wont be perfect I think, but it will certainly help most of the time.

**Comment by cartermp** [comment]
Love it!

**Pull request merged**